internal/goreleaser/goreleaser.go

```diff
@@ -104,38 +52,32 @@ func RenderConfig(cfg core.Configuration) {
 		binaryName = cfg.GoReleaser.BinaryName
 	}
 
-	if len(cfg.Golang.LdFlags) > 0 {
-		var names []string
-		for key := range cfg.Golang.LdFlags {
-			names = append(names, key)
-		}
-		sort.Strings(names)
-
-		for _, name := range names {
-			value := cfg.Golang.LdFlags[name]
-			ldflags += fmt.Sprintf(strings.ReplaceAll(`
-			- %s={{.Env.%s}}`, "\t", "  "), name, value)
-		}
-	}
-
+	var (
+		metadataURL *url.URL
+		err         error
+	)
 	if !strings.HasPrefix(cfg.Metadata.URL, "https://github.com/") {
-		metadataURL, err := url.Parse(cfg.Metadata.URL)
+		metadataURL, err = url.Parse(cfg.Metadata.URL)
 		if err != nil {
 			logg.Fatal("Metadata.URL is not a parsable URL: %w", err)
 		}
-
-		githubURLs = fmt.Sprintf(strings.ReplaceAll(`
-github_urls:
-	api: https://%[1]s/api/v3/
-	upload: https://%[1]s/api/uploads/
-	download: https://%[1]s/
-`, "\t", "  "), metadataURL.Host)
 	}
 
-	goreleaserFile := fmt.Sprintf(goreleaserTemplate, nameTemplate, format, files, binaryName, cfg.Binaries[0].Name, ldflags, cfg.Binaries[0].FromPackage, githubURLs)
+	t := template.Must(template.New("goreleaser.yaml").Parse(goreleaserTemplate))
+	var goreleaserFile bytes.Buffer
+	must.Succeed(t.Execute(&goreleaserFile, map[string]any{
```

Comment:
Nitpick: Technically, `goreleaserFile` can be an `os.File`, which would avoid the buffer.